tests/queries/0_stateless/02596_build_set_and_remote.sql

```diff
@@ -0,0 +1,12 @@
+-- {echoOn}
+SELECT arrayExists(x -> (x IN (SELECT '2')), [2]) FROM system.one;
+
+SELECT arrayExists(x -> (x IN (SELECT '2')), [2]) FROM remote('127.0.0.{2,3}', system.one);
+
+SELECT arrayExists(x -> (x IN (SELECT '2')), [2]) FROM remote('127.0.0.{2,3}', system.one) GROUP BY NULL;
+
+SELECT arrayExists(x -> (x IN (SELECT '2')), [2]) FROM remote('127.0.0.{2,3}', system.one) GROUP BY 1;
+
+SELECT arrayExists(x -> (x IN (SELECT '2')), [2]) FROM remote('127.0.0.{2,3}', system.one) GROUP BY 'A';
+
```

Comment:
Minified repro from https://s3.amazonaws.com/clickhouse-test-reports/49418/8b0f6edb4269d3e6ebbcddf45d47a788beab9c38/fuzzer_astfuzzermsan/report.html

```suggestion

SELECT 1 IN ( SELECT 1 ) FROM remote('127.0.0.{1,2}', system.one) GROUP BY dummy
```